### PR TITLE
Add mfact dispatcher

### LINKED ISSUE #3292: Changing mfact using keyword doesn't update until layout change
### Hyprland Version

v0.29.1

### Bug or Regression?

Bug

### Description

I have a script that I want to use to toggle the `mfact` of the master window in the master layout. Unless I include the lines to switch the layout type, the master window does not update. `getoption` does however return the correct value. Here is the current script:

```
#!/bin/sh

if [ $(hyprctl getoption master:mfact -j | jq -r ".float") == "0.330000" ]
then
    hyprctl keyword master:mfact 0.5
    hyprctl keyword general:layout "dwindle"
    hyprctl keyword general:layout "master"
else
    hyprctl keyword master:mfact 0.33
    hyprctl keyword general:layout "dwindle"
    hyprctl keyword general:layout "master"
fi
```

Ideally the `hyprctl keyword general:layout ...` lines shouldn't be necessary.

### How to reproduce

`hyprctl keyword master:mfact 0.5`

nothing happens.

### Crash reports, logs, images, videos

_No response_

### PR DESCRIPTION
**Pull request by thejch**
#### Describe your PR, what does it fix/add?
Fixes #3292 . Add a dispatcher for changing mfact.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
no

#### Is it ready for merging, or does it need work?
ready

### CODE REVIEW

File: src/layout/MasterLayout.cpp [hunk 1]
```diff
@@ -1234,6 +1234,13 @@ std::any CHyprMasterLayout::layoutMessage(SLayoutMessageHeader header, std::stri
         runOrientationCycle(header, nullptr, -1);
     } else if (command == "orientationcycle") {
         runOrientationCycle(header, &vars, 1);
+    } else if (command == "mfact") {
+        if (vars.size() >= 2) {
+            for (auto& nd : m_lMasterNodesData) {
+                if (nd.isMaster)
+                    nd.percMaster = std::clamp(std::stof(vars[1]), 0.05f, 0.95f);
```
Comment (vaxerski) [inline src/layout/MasterLayout.cpp:1241]: every use of std::stof has to be put in a try{} block.

Secondly, no need to `std::stof` every time.

before the for loop:
```cpp
float ejfurngnurg = 0;
try {
   ejfurngnurg = std::stof(vars[1]);
} catch (std::exception& e) {
   Debug::log(LOG, "OOPSIE WOOPSIE LMAO XD {}", e.what());
}
```

Comment (thejch) [reply src/layout/MasterLayout.cpp]: fixed

File: src/layout/MasterLayout.cpp [hunk 2]
```diff
@@ -1234,6 +1234,20 @@ std::any CHyprMasterLayout::layoutMessage(SLayoutMessageHeader header, std::stri
         runOrientationCycle(header, nullptr, -1);
     } else if (command == "orientationcycle") {
         runOrientationCycle(header, &vars, 1);
+    } else if (command == "mfact") {
+        if (vars.size() >= 2) {
+            float newMfact = 0;
+            try {
+                newMfact = std::stof(vars[1]);
+            } catch (std::exception& e) {
+                Debug::log(LOG, "Argument is invalid: {}", e.what());
```
Comment (vaxerski) [inline src/layout/MasterLayout.cpp:1243]: why isnt this an ERR?

**Review by vaxerski — COMMENTED** [review]
apart from that, all good

File: src/layout/MasterLayout.cpp (hunk 2 above)
Comment (thejch) [reply src/layout/MasterLayout.cpp]: right. fixed

**Review by vaxerski — APPROVED** [review]
lgtm

I can feel it coming in the air tonight

**Comment by vaxerski** [comment]
wiki mr please before merging

**Pull request merged**